### PR TITLE
Add retries to DownloadPackage task.

### PR DESCRIPTION
**Pull request by omeshp**

### CODE REVIEW

**Comment by omeshp** [comment]
#8934

File: Tasks/DownloadPackageV0/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -18,5 +18,7 @@
   "loc.messages.StartingDownloadOfPackage": "Starting download of NuGet package %s to location %s",
   "loc.messages.ExtractingNugetPackage": "Extracting NuGet package %s to directory %s",
   "loc.messages.PackageTypeNotSupported": "Only NuGet packages types can be downloaded using this task.",
-  "loc.messages.ExtractionFailed": "Failed to extract package with error %s"
+  "loc.messages.ExtractionFailed": "Failed to extract package with error %s",
+  "loc.messages.RetryingOperation": "Error: in %s, so retrying => retries pending  : %s",
```
Comment (Lovakumar) [inline Tasks/DownloadPackageV0/Strings/resources.resjson/en-US/resources.resjson]: remove space between pending and column.

File: Tasks/DownloadPackageV0/download.ts [hunk 2]
```diff
@@ -30,8 +30,11 @@ async function main(): Promise<void> {
 	var credentialHandler = vsts.getBearerHandler(accessToken);
 	var vssConnection = new vsts.WebApi(collectionUrl, credentialHandler);
 	var coreApi = vssConnection.getCoreApi();
+	var retryLimit = parseInt(tl.getVariable("VSTS_HTTP_RETRY")) ? parseInt(tl.getVariable("VSTS_HTTP_RETRY")) : 4;
```
Comment (Lovakumar) [inline Tasks/DownloadPackageV0/download.ts]: Kalyan found a bug with this line of code. It wasn't handling zero well. Refer to recent check-ins on Jenkins task.

File: Tasks/DownloadPackageV0/task.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "version": {
         "Major": 0,
         "Minor": 1,
-        "Patch": 17
+        "Patch": 18
```
Comment (kasubram) [inline Tasks/DownloadPackageV0/task.json]: Everywhere else we follow milestone number for minor version. Should we change this?

**Pull request merged**